src/service/compact/retention.rs

```diff
@@ -59,27 +102,64 @@ pub async fn delete_by_stream(
         lifecycle_end
     );
 
-    // Hack for 1970-01-01
-    if lifecycle_start.le("1970-01-01") {
-        let lifecycle_end = created_at + Duration::try_days(1).unwrap();
-        let lifecycle_end = lifecycle_end.format("%Y-%m-%d").to_string();
-        return db::compact::retention::delete_stream(
+    // Create tasks with break on red days
+    let lifecycle_start = format!("{}T00:00:00Z", lifecycle_start);
+    let lifecycle_end = format!("{}T00:00:00Z", lifecycle_end);
+    let start = DateTime::parse_from_rfc3339(&lifecycle_start)?.with_timezone(&Utc);
+    let end = DateTime::parse_from_rfc3339(&lifecycle_end)?.with_timezone(&Utc);
+    if start.ge(&end) {
+        return Ok(0); // created_at is after lifecycle_end, just skip
+    }
+
+    let original_time_range = TimeRange::new(
+        start.timestamp_nanos_opt().expect("valid timestamp") / 1000,
+        end.timestamp_nanos_opt().expect("valid timestamp") / 1000,
+    );
+
+    let mut final_time_ranges = vec![];
+    red_days.iter().for_each(|red_day| {
+        let _ranges_added = populate_time_ranges_for_deletion(
+            &mut final_time_ranges,
+            red_day,
+            &original_time_range,
+        );
+    });
```

Comment:
if the red data has cross date, does it work? for example:
- 2024-12-10 - 2024-12-20
- 2024-12-15 - 2024-12-22

Reply:
yeah, I can see it creating issues. This makes it a bit more complex. Will fix.